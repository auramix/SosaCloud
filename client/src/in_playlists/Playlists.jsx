import styled from 'styled-components';
import React from 'react';
import SinglePlaylist from './SinglePlaylist.jsx';
import List from '../styled-comps/RelatedTrackList.jsx';

export default class Playlists extends React.Component {
  constructor(props) {
    super(props);
    this.state = { playlists: [], artistPopUpOpen: false  };
  }

  componentDidMount() {
    let id = this.props.id || Math.ceil(Math.random() * 100);

    fetch(`/api/playlists/${id}`, {
      method: 'GET'
    })
      .then(data => data.json())
      .then(jsonData => {
        if (Array.isArray(jsonData)) {
          console.log('This is our jsonData: ', jsonData);
          this.setState({ playlists: jsonData })
        }
      })
      .then(() => {
        if (this.state.playlists.length) {
          this.props.renderPlaylists();
        }
      })
      .catch((err) => {
        console.log('***Client componentDidMount fetch error in Playlists component***, ', err);
      })
  }

  
  render() {
    let playlists = this.state.playlists;
    if (playlists.length > 0) {
      playlists = playlists.map(playlist => <SinglePlaylist key={playlist.id} playlist={playlist} />);
    }
    return (
        <List>
          {playlists}
        </List>
    );
  }
}