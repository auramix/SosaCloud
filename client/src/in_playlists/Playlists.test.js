import Playlists from './Playlists.jsx';

describe('Playlists', () => {
  // Snapshot test
  it('Has expected snapshot output', () => {
    const wrapper = shallow( <
      Playlists renderPlaylists = {
        () => {}
      }
      />, {disableLifecycleMethods: true}
    );
    expect(wrapper).toMatchSnapshot();
  })

});