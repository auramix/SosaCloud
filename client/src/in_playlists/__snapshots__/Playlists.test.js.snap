// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playlists Has expected snapshot output 1`] = `<styled.ul />`;
